### PR TITLE
Use the DRY kubernetes client flags in sinker

### PR DESCRIPTION
**Pull request by stevekuznetsov**
This doesn't change the flag names, just refactors how they're set.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @cjwagner @fejta 
/cc @BenTheElder @krzyzacy

### CODE REVIEW

File: prow/cmd/sinker/main.go [hunk 1]
```diff
@@ -19,39 +19,56 @@ package main
...
+	fs.StringVar(&o.configPath, "config-path", "/etc/config/config.yaml", "Path to config.yaml.")
+	fs.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
+
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether or not to make mutating API calls to Kubernetes.")
+	o.kubernetes.AddFlags(fs)
+	fs.Parse(os.Args[1:])
 	return o
 }
+
+func (o *options) Validate() error {
```
Comment (fejta) [inline prow/cmd/sinker/main.go]: Not obvious where Validate is called...

Comment (stevekuznetsov) [reply prow/cmd/sinker/main.go]: :man_facepalming:

**Comment by stevekuznetsov** [comment]
@fejta fixed

**Comment by stevekuznetsov** [comment]
@krzyzacy @fejta added the `ANNOUNCEMENTS.md` here

File: prow/cmd/sinker/main.go [hunk 2]
```diff
@@ -19,41 +19,62 @@ package main
...
-	flag.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
-	flag.StringVar(&o.buildCluster, "build-cluster", "", "Path to kube.Cluster YAML file. If empty, uses the local cluster.")
-	flag.StringVar(&o.buildClusterKubeconfig, "kubeconfig", "", "Path to kubeconfig with build cluster credentials. If empty, defaults to in-cluster config.")
-	flag.Parse()
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.runOnce, "run-once", false, "If true, run only once then quit.")
+	fs.StringVar(&o.configPath, "config-path", "/etc/config/config.yaml", "Path to config.yaml.")
+	fs.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
+
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether or not to make mutating API calls to Kubernetes.")
```
Comment (fejta) [inline prow/cmd/sinker/main.go]: confirm?

Comment (stevekuznetsov) [reply prow/cmd/sinker/main.go]: Just cargo-culting today, can change in the future

**Pull request merged**